appservice/src/deploy/getDeployNode.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AzExtTreeItem } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { getWorkspaceSetting, updateWorkspaceSetting } from '../utils/settings';
+import { AppSource, IDeployContext } from './IDeployContext';
+
+export async function getDeployNode<T extends AzExtTreeItem>(context: IDeployContext, arg1: unknown, arg2: unknown, expectedContextValue: string | string[]): Promise<T> {
```

Comment:
Because these args are coming from VS Code and we have little control over them. For example, we had a bug in deploy where VS Code was passing an empty array as arg2 and we weren't handling that correctly: https://github.com/microsoft/vscode-azureappservice/pull/1420

Reply:
> I'm not entirely sure why getDeployNode needs two different args parameters since both are going to be used to find the deployNode (in different ways). It seems like the function outside of getDeployNode can handle which argument should be passed in?

Added a bit of docs. I want this function to take in whatever gets passed to the command itself because (as linked above) we've had problems where VS Code passes arguments that we didn't expect and I want to share logic to handle that.